x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -365,11 +371,15 @@ public void testNodesInfo() throws IOException {
             // remote cluster is not reported in transport profiles
             assertThat(ObjectPath.eval("transport.profiles", node), anEmptyMap());
 
-            final List<String> boundAddresses = ObjectPath.eval("remote_cluster_server.bound_address", node);
-            assertThat(boundAddresses, notNullValue());
-            assertThat(boundAddresses, not(empty()));
-            final String publishAddress = ObjectPath.eval("remote_cluster_server.publish_address", node);
-            assertThat(publishAddress, notNullValue());
+            if (Boolean.parseBoolean(ObjectPath.eval("settings.remote_cluster_server.enabled", node))) {
+                final List<String> boundAddresses = ObjectPath.eval("remote_cluster_server.bound_address", node);
+                assertThat(boundAddresses, notNullValue());
+                assertThat(boundAddresses, not(empty()));
+                final String publishAddress = ObjectPath.eval("remote_cluster_server.publish_address", node);
+                assertThat(publishAddress, notNullValue());
+            } else {
+                assertThat(ObjectPath.eval("remote_cluster_server", node), nullValue());
+            }
```

Comment:
This actually helps me to spot a test bug. Settings using the static fields such as `NODE2_RCS_SERVER_ENABLED` needs to use the method with the `Supplier` variant. Otherwise, the settings are set before the static fields can be initialized. In this case, they are always false. I have now fixed it accordingly. Thanks!